### PR TITLE
Add How to Detect and Annotate guide

### PR DESCRIPTION
**Pull request by capjamesg**
This PR adds a `How to Detect and Annotate` guide to the `supervision` quickstart.

### CODE REVIEW

**Comment by onuralpszr** [comment]
I think we should also add "full code"  in the end of it as well.

File: docs/how_to/detect_and_annotate.md [hunk 1]
```diff
@@ -1 +1,63 @@
-🚧 Page under construction.
+## How to Detect and Annotate
+
+Supervision enables you to annotate predictions from object detection and segmentation models in a few lines of code.
+
+The `sv.BoxAnnotator()` class lets you annotate images with bounding boxes, while the `sv.MaskAnnotator()` class lets you annotate images with segmentation masks. `sv.MaskAnnotator()` is a drop-in replacement for `sv.BoxAnnotator()`, so you don't need to update any other code if you want to switch between the two.
```
Comment (SkalskiP) [inline docs/how_to/detect_and_annotate.md:5]: A good start! I want mentions of supervision API elements to link to docs.

Comment (SkalskiP) [reply docs/how_to/detect_and_annotate.md:5]: Also, if we can, lets use `sv.BoxAnnotator` not `sv.BoxAnnotator()`

File: docs/how_to/detect_and_annotate.md [hunk 2]
```diff
@@ -1 +1,63 @@
...
+
+Finally, we can display the annotated image:
+
+```python
+sv.plot_image(annotated_image)
+```
+
+Here is the result:
+
+![Predictions plotted on an image](https://media.roboflow.com/supervision_annotate.png)
```
Comment (SkalskiP) [inline docs/how_to/detect_and_annotate.md:63]: Let's use horizontal images, please. I try keep the docs concise.

File: docs/how_to/detect_and_annotate.md [hunk 3]
```diff
@@ -1 +1,63 @@
...
+import cv2
+
+model = YOLO("yolov8n.pt")
+
+results = model("image.jpg")
+```
+
+In this code, we import `supervision`, `ultralytics`, and `cv2` (OpenCV). We then load a YOLOv8 object detection model, then run inference on an image called `image.jpg`. The `results` variable contains the predictions from the model.
+
+## Load Predictions into Supervision
```
Comment (SkalskiP) [inline docs/how_to/detect_and_annotate.md:25]: Please mention other connectors, especially `from_roboflow`. We probably need to use YOLOv8 as an example, but let's show other options.

File: docs/how_to/detect_and_annotate.md [hunk 4]
```diff
@@ -1 +1,63 @@
...
+
+The `sv.BoxAnnotator()` class lets you annotate images with bounding boxes, while the `sv.MaskAnnotator()` class lets you annotate images with segmentation masks. `sv.MaskAnnotator()` is a drop-in replacement for `sv.BoxAnnotator()`, so you don't need to update any other code if you want to switch between the two.
+
+In this guide, we run inference on an Ultralytics YOLOv8 object detection model and plot the predictions with the `sv.BoxAnnotator()` class.
+
+## Load a Model and Retrieve Predictions
+
+First, we need to retrieve predictions from a model. For this guide, we will retrieve predictions from a YOLOv8 model. See a list of all models from which you can load predictions into supervision.
+
+```python
```
Comment (SkalskiP) [inline docs/how_to/detect_and_annotate.md:13]: Let's change this example to use `image = cv2.imread("image.jpg")`.

File: docs/how_to/detect_and_annotate.md [hunk 5]
```diff
@@ -1 +1,63 @@
...
+import supervision as sv
+from ultralytics import YOLO
+import cv2
+
+model = YOLO("yolov8n.pt")
+
+results = model("image.jpg")
```
Comment (SkalskiP) [inline docs/how_to/detect_and_annotate.md:20]: If I'm not mistaken we need to add `[0]` at the end.

File: docs/how_to/detect_and_annotate.md (hunk 2 above)
Comment (SkalskiP) [reply docs/how_to/detect_and_annotate.md:63]: Also, in the result example you show labels, but the code snippet does not match the result.

File: docs/how_to/detect_and_annotate.md [hunk 6]
```diff
@@ -1 +1,63 @@
...
+```
+
+In this code, we import `supervision`, `ultralytics`, and `cv2` (OpenCV). We then load a YOLOv8 object detection model, then run inference on an image called `image.jpg`. The `results` variable contains the predictions from the model.
```
Comment (SkalskiP) [inline docs/how_to/detect_and_annotate.md:23]: Just a thought: people don't have `image.jpg`. Maybe we could make our docs in such a way that we use some hosted images.

**Review by SkalskiP — CHANGES_REQUESTED** [review]

**Comment by SkalskiP** [comment]
@onuralpszr added it with my update. Merging! 🔥

**Pull request merged**